stdlib/public/SwiftShims/DispatchOverlayShims.h

```diff
@@ -215,8 +215,8 @@ static inline unsigned int
 _swift_dispatch_data_apply(
     __swift_shims_dispatch_data_t data,
     __swift_shims_dispatch_data_applier SWIFT_DISPATCH_NOESCAPE applier) {
-  return dispatch_data_apply(data, ^bool(dispatch_data_t data, size_t off, const void *loc, size_t size){
-    return applier(data, off, loc, size);
+  return dispatch_data_apply((dispatch_data_t)data, ^bool(dispatch_data_t data, size_t off, const void *loc, size_t size){
```

Comment:
An alternative solution would be to update `dispatch_data_apply` to take a `__swift_shims_dispatch_data_t`. That would modify a public function to accept a private type so, I figured this solution would be better.